admob/src/ios/rewarded_ad_internal_ios.mm

```diff
@@ -30,23 +30,148 @@
     ad_load_callback_data_(nil), rewarded_ad_(nil),
     parent_view_(nil), rewarded_ad_delegate_(nil) {}
 
-RewardedAdInternalIOS::~RewardedAdInternalIOS() { }
+RewardedAdInternalIOS::~RewardedAdInternalIOS() {
+  firebase::MutexLock lock(mutex_);
+  // Clean up any resources created in RewardedAdInternalIOS.
+  Mutex mutex(Mutex::kModeNonRecursive);
```

Comment:
Is this the model we use for other things that block until the destruction is complete? Or do we use a Semaphore?